src/Coq.hs

```diff
@@ -0,0 +1,410 @@
+{-
+ -
+ - coq backend for act
+ -
+ - unsupported features:
+ - + bytestrings
+ - + external storage
+ - + specifications for multiple contracts
+ -
+ -}
+
+{-# Language OverloadedStrings #-}
+{-# LANGUAGE GADTs #-}
+
+module Coq where
+
+import qualified Data.Map.Strict    as M
+import qualified Data.List.NonEmpty as NE
+import qualified Data.Text          as T
+import Data.Either (rights)
+import Data.List (find, groupBy)
+import Control.Monad.State
+
+import EVM.ABI
+import EVM.Solidity (SlotType(..))
+import Syntax
+import RefinedAst
+
+type Store = M.Map Id SlotType
+type Fresh = State Int
+
+header :: T.Text
+header =
+  "(* --- GENERATED BY ACT --- *)\n\n\
```

Comment:
here (and in general throughout the code) it might be nice to use `unlines` to make things more readable

Reply:
patched in https://github.com/ethereum/act/commit/c5c6e3e7ce127f2a7a165985c8165d00e69f23c6. coq isn't newline sensitive in a lot of cases, so I only used it in places where it makes sense to emphasize the line structure.